doc/dev/tests-advanced.md

```diff
@@ -74,114 +78,61 @@ class TestTablesUnit(TablesTestMixin):
 
-### Example 3: Cached Preparer Usage
 ```python
-import os
-import pytest
-
-from azure.core.exceptions import ResourceExistsError
-from azure.data.tables import TableServiceClient
-from devtools_testutils import (
-    AzureTestCase,
-    CachedResourceGroupPreparer,
-    CachedStorageAccountPreparer
-)
-
-class ExampleStorageTestCase(AzureTestCase):
-
-    @CachedResourceGroupPreparer(name_prefix="storagetest")
-    @CachedStorageAccountPreparer(name_prefix="storagetest")
-    def test_create_table(self, resource_group, location, storage_account, storage_account_key):
-        account_url = self.account_url(storage_account, "table")
-        client = self.create_client_from_credential(TableServiceClient, storage_account_key, account_url=account_url)
-
-        valid_table_name = "validtablename"
-        table = client.create_table(valid_table_name)
-
-        assert valid_table_name == table.table_name
-
-    @CachedResourceGroupPreparer(name_prefix="storagetest")
-    @CachedStorageAccountPreparer(name_prefix="storagetest")
-    def test_create_table_if_exists (self, resource_group, location, storage_account, storage_account_key):
-        account_url = self.account_url(storage_account, "table")
-        client = self.create_client_from_credential(TableServiceClient, storage_account_key, account_url=account_url)
-
-        valid_table_name = "validtablename"
-        with pytest.raises(ResourceExistsError):
-            table = client.create_table(valid_table_name)
+@pytest.fixture(scope="session")
+def setup_fixture():
+    credential = get_credential()  # needs to be manually implemented; we can't use AzureRecordedTestCase.get_credential
+    client = ServiceClient("...", credential)
+    client.set_up_resource()
+    yield
+
+@pytest.mark.usefixtures("setup_fixture")
+class TestService(AzureRecordedTestCase):
+    ...
 ```
 
-The first test is the same as above, the second test tries to create a table that already exists and asserts that the correct type of error is raised in response. These tests use cached preparers unlike the previous example.
-
-Notes:
-1. The cached preparers here will first look to see if an existing resource group or storage account exists with the given parameters, in this case the `name_prefix`. For more information on what parameters differentiate a new resource group or storage account look for the `self.set_cache()` method in the preparer source code [here](https://github.com/Azure/azure-sdk-for-python/blob/main/tools/azure-sdk-tools/devtools_testutils/storage_testcase.py#L49). The advantage to using a cached preparer is the time saver to re-using the same resource instead of creating a new resource for each test. However, this can increase the possibility that you have to be more exact about cleaning up the entities created in between test runs.
+By requesting a fixture from the test class, the fixture will execute before any tests in the class do. Fixtures are the
```

Comment:
Nit: I think we should mention someplace that the setup fixtures should exist in the `conftest.py` for the test.\. xunit style setup implies in your test class, but this one doesn't really tell folks where the code should exist.